### PR TITLE
[tests-only][full-ci] add test to share project drive's resource to a federated user

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds ocm test to check activities of a resource from personal drive shared to a federated user.
``` feature
Scenario: local user shares resources from project space to federation user
```

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/10073

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiOcm/share.feature [hunk 1]
```diff
@@ -100,6 +100,96 @@ Feature: an user shares resources usin ScienceMesh application
       | folderToShare |
       | textfile.txt  |
 
+
+  Scenario: local user shares resources from project space to federation user
```
Comment (anon-pradip) [inline tests/acceptance/features/apiOcm/share.feature:104]: IMHO,
```suggestion
  Scenario: check activity after sharing the resource from project space of local user to federation user
```
will make the scenario more clear? :thinking:

Comment (prashant-gurung899) [reply tests/acceptance/features/apiOcm/share.feature:104]: Yeah, makes sense but just following the same pattern as other scenarios in that feature file.

File: tests/acceptance/features/apiOcm/share.feature [hunk 2]
```diff
@@ -100,6 +100,85 @@ Feature: an user shares resources usin ScienceMesh application
...
+      | resource        | folderToShare                 |
+      | space           | projectSpace                  |
+      | sharee          | Brian                         |
+      | shareType       | user                          |
+      | permissionsRole | Viewer                        |
+      | federatedServer | @federation-ocis-server:10200 |
+    Then the HTTP status code should be "200"
+    When using server "REMOTE"
+    And user "Brian" lists the shares shared with him without retry using the Graph API
+    Then the HTTP status code should be "200"
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:124]: scenario doesn't align with the PR title.

File: tests/acceptance/features/apiOcm/share.feature (hunk 1 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiOcm/share.feature:104]: sorry but where activity api has been hit?

**Comment by amrita-shrestha** [comment]
@saw-jan why test scenario have multiple when for ocm. Any specific reason. Because it seems out of our current practice

**Review by amrita-shrestha — CHANGES_REQUESTED** [review]
please look for spelling error in
```
Feature: an user shares resources usin ScienceMesh application
```

File: tests/acceptance/features/apiOcm/share.feature (hunk 1 above)
Comment (prashant-gurung899) [reply tests/acceptance/features/apiOcm/share.feature:104]: We are not hitting activity api as we're just listing the shares not checking the activity.

Comment (saw-jan) [reply tests/acceptance/features/apiOcm/share.feature:104]: was just the misleading PR title - which has been changed

File: tests/acceptance/features/apiOcm/share.feature [hunk 3]
```diff
@@ -100,6 +100,85 @@ Feature: an user shares resources usin ScienceMesh application
...
+              "required": [
+                "@UI.Hidden",
+                "@client.synchronize",
+                "createdBy",
+                "name"
+              ],
+              "properties": {
+                "@UI.Hidden": {
+                  "type": "boolean",
+                  "enum": [ false ]
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:146]:
```suggestion
                  "const": false
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 4]
```diff
@@ -100,6 +100,85 @@ Feature: an user shares resources usin ScienceMesh application
...
+                },
+                "@client.synchronize": {
+                  "type": "boolean",
+                  "enum": [ false ]
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:150]:
```suggestion
                  "const": false
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 5]
```diff
@@ -100,6 +100,85 @@ Feature: an user shares resources usin ScienceMesh application
...
+                  "required": [ "user" ],
+                  "properties": {
+                    "user": {
+                      "type": "object",
+                      "required": [ "displayName", "id" ],
+                      "properties": {
+                        "displayName": {
+                          "type": "string",
+                          "const": "Alice Hansen"
+                        },
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:163]:
```suggestion
                          "const": "Alice Hansen"
                        },
```

Comment (saw-jan) [reply tests/acceptance/features/apiOcm/share.feature:163]: because `const` asserts both the type and the value

**Review by saw-jan — APPROVED** [review]
:+1:

**Comment by saw-jan** [comment]
> @saw-jan why test scenario have multiple when for ocm. Any specific reason. Because it seems out of our current practice

In some cases, it should be fine unless you misuse it. It is always good to have atomic test scenarios but we also have to consider the  growing number of tests
Multiple When can be refactored into Given, Then but in the coming PRs

**Review by amrita-shrestha — APPROVED** [review]

**Pull request merged**